### DISCUSSION TITLE
Changing route params with RouterLink component

### DISCUSSION THREAD

**Discussion by PindaPixel** [vuejs/router#2168] [Help and Questions] [answered]
I have a case where I have two links in the sidebar:
```typescript
[
    {
        name: "check/reports/kpi/:type/:id?",
        params: {
            type: "project",
            id: undefined,
        },
    },
    {
        name: "check/reports/kpi/:type/:id?",
        params: {
            type: "organisation",
            id: undefined,
        },
    }
]
```
When I am situated on the URL `check/reports/kpi/project/1` and click on the second link that I want to navigate to `check/reports/kpi/organisation`, instead it navigates to `check/reports/kpi/organisation/1`. This is problematic, because the ID in this case is not interchangeable. I would think setting id to `undefined` or `null` explicitly would give me the desired behaviour. Is there any other way to achieve this?

**Comment by posva** [comment, marked as answer]
Passing an empty string should also work but explicitly passing `null` or undefined shouldn't implicitly preserve existing params either

**Reply by PindaPixel** [reply]
Then this is either an error or my end or a bug that was introduced in a recent release, I'll investigate on my end to see what I can find.

**Reply by PindaPixel** [reply]
Was unable to reproduce the issue in a minimal example. It works as you described it should: https://stackblitz.com/edit/vitejs-vite-flfizf?file=src%2Fmain.js,src%2FHome.vue,src%2FNested.vue&terminal=dev

So it's likely a fault in my configuration.

**Reply by PindaPixel** [reply]
For anyone who wishes to know, I've updated the stackblitz above. The problem was that my route was defined as follows:
```typescript
{
    path: '/',
    component: Home,
    children: [
      {
        path: ':resource',
        children: [
          {
            path: ':id?',
            children: [
              {
                path: '',
                name: 'test',
                component: Nested,
              },
            ],
          },
        ],
      },
    ],
  },
```
While it should have been:
```typescript
{
    path: '/',
    component: Home,
    children: [
      {
        path: ':resource',
        children: [
          {
            path: ':id?',
            name: 'test',
            component: Nested,
          },
        ],
      },
    ],
  },
```
